### ISSUE TITLE
Product Flavor tasks don't find their tests properly

### ISSUE THREAD

**Issue by Adambl4** [mannodermaus/android-junit5#25]
[Subject](https://github.com/Adambl4/Spek-and-AGP3/blob/master/app/build.gradle#L23-L29)

**Comment by mannodermaus** [comment]
What error are you facing when the product flavor is active? Starting with AGP3, each product flavor has to be assigned a `dimension`, which you declared, but didn't assign to the flavor (it should say `dev { dimension "tier" }`).

[Here is an example](https://github.com/mannodermaus/android-junit5/blob/9de6cdea49b517c2806c7acee43c45895df44458/android-junit5/src/testAgp3x/groovy/de/mannodermaus/gradle/plugins/android_junit5/AGP3PluginSpec.groovy#L21-L26) from the plugin's unit tests. Please let me know if the error persists beyond that, and post a stack trace as well!

**Comment by Adambl4** [comment]
The problem is that it doesn't found the Spek test.
This is the result of `./gradlew junitPlatformTestDevDebug` with an existing product flavour
```
Test run finished after 28 ms
[         3 containers found      ]
[         0 containers skipped    ]
[         3 containers started    ]
[         0 containers aborted    ]
[         3 containers successful ]
[         0 containers failed     ]
[         **0 tests found**           ]
[         0 tests skipped         ]
[         0 tests started         ]
[         0 tests aborted         ]
[         0 tests successful      ]
[         0 tests failed          ]
```
And this is of `./gradlew junitPlatformTest` without a product flavour
```
Test run finished after 32 ms
[         4 containers found      ]
[         0 containers skipped    ]
[         4 containers started    ]
[         0 containers aborted    ]
[         4 containers successful ]
[         0 containers failed     ]
[         **1 tests found**           ]
[         0 tests skipped         ]
[         1 tests started         ]
[         0 tests aborted         ]
[         0 tests successful      ]
[         1 tests failed          ]
```

**Comment by mannodermaus** [comment]
Thanks for elaborating! This looks like an issue with how the different source paths are fed into the flavor-aware sub tasks. I'll keep you posted.

**Comment by mannodermaus** [comment]
Actually the problem wasn't how the JUnit tasks were set up, but rather where the intermediate Kotlin classes were copied to. If you used a Java test instead, it would have been picked up properly. After some writing, I have now introduced functional tests which will aid in verifying the behavior of the plugin with actual unit test sources.

On the latest snapshot of the plugin, `1.0.12-SNAPSHOT`, this issue has been fixed, and it will be part of the next proper release, also due very shortly.

Thanks for reporting!
